### PR TITLE
modifying SyncBN doc for FP16 use case

### LINKED ISSUE #13976: Problem of exporting FP16 SyncBN model.
## Description
This is a problem when exporting fp16 model containing SyncBN.

```
import mxnet as mx

mx.random.seed(42)

def data_xform(data):
    """Move channel axis to the beginning, cast to float32, and normalize to [0, 1]."""
    return mx.nd.moveaxis(data, 2, 0).astype('float32') / 255

train_data = mx.gluon.data.vision.MNIST(train=True).transform_first(data_xform)
val_data = mx.gluon.data.vision.MNIST(train=False).transform_first(data_xform)

batch_size = 2000
train_loader = mx.gluon.data.DataLoader(train_data, shuffle=True, batch_size=batch_size)
val_loader = mx.gluon.data.DataLoader(val_data, shuffle=False, batch_size=batch_size)

net = mx.gluon.nn.HybridSequential()
with net.name_scope():
    net.add(mx.gluon.nn.Conv2D(64, (3, 3)))
    net.add(mx.gluon.contrib.nn.SyncBatchNorm())
    net.add(mx.gluon.nn.Conv2D(64, (3, 3)))
    net.add(mx.gluon.contrib.nn.SyncBatchNorm())
    net.add(mx.gluon.nn.Dense(128))
    net.add(mx.gluon.nn.Dense(10))
    net.hybridize()
    net.cast('float16')
print('finish build the network')

ctx =  [mx.gpu(int(id)) for id in [0,1,2,3,4,5,6,7]]
net.initialize(mx.init.Normal(0.01), ctx=ctx)
print('finish initializing')

trainer = mx.gluon.Trainer(
    params=net.collect_params(),
    optimizer='sgd',
    optimizer_params={'learning_rate': 0.04, 'multi_precision':True},
)

loss_function = mx.gluon.loss.SoftmaxCrossEntropyLoss()
metric = mx.metric.Accuracy()

num_epochs = 10
for epoch in range(num_epochs):
    print('start training')
    for inputs, labels in train_loader:
        data = mx.gluon.utils.split_and_load(inputs, ctx_list=ctx, batch_axis=0)
        label = mx.gluon.utils.split_and_load(labels, ctx_list=ctx, batch_axis=0)
        losses = []
        with mx.autograd.record():
            for X, Y in zip(data, label):
                outputs = net(X.astype('float16'))
                loss = loss_function(outputs.astype('float32'), Y)
                losses.append(loss)

        for l in losses:
            l.backward()
        trainer.step(batch_size=inputs.shape[0])

    net.export('mnist_syncbn_fp16.params')
```

## Error Message:
```
File "/home/ubuntu/Workspace/incubator-mxnet/python/mxnet/gluon/block.py", line 900, in export
    ndarray.save('%s-%04d.params'%(path, epoch), arg_dict)
  File "/home/ubuntu/Workspace/incubator-mxnet/python/mxnet/ndarray/utils.py", line 273, in save
    keys))
  File "/home/ubuntu/Workspace/incubator-mxnet/python/mxnet/base.py", line 255, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: [18:43:24] include/mxnet/././tensor_blob.h:203: Check failed: mshadow::DataType<DType>::kFlag == type_flag_ TBlob.get_with_shape: data type do not match specified type.Expected: 2 v.s. given 0
```

### PR DESCRIPTION
**Pull request by mseth10**
## Description ##
Updating docs to add instructions for MXNet users on SyncBN FP16 support and possible workarounds. This fixes #13976 .

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] SyncBN doc

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [pr-awaiting-review, Doc]

**Comment by vandanavk** [comment]
@mxnet-label-bot update [pr-awaiting-merge, Doc]

@aaronmarkham for review/merge

File: python/mxnet/gluon/contrib/nn/basic_layers.py [hunk 1]
```diff
@@ -165,7 +165,10 @@ class SyncBatchNorm(BatchNorm):
 
     Standard BN [1]_ implementation only normalize the data within each device.
     SyncBN normalizes the input within the whole mini-batch.
-    We follow the sync-onece implmentation described in the paper [2]_.
+    We follow the implementation described in the paper [2]_.
+
+    Note: Current implementation of SyncBN does not support FP16 training.
```
Comment (lupesko) [inline python/mxnet/gluon/contrib/nn/basic_layers.py]: So does it not support training or inference?
You say training in this line, and refer to inference in the next line.

Comment (mseth10) [reply python/mxnet/gluon/contrib/nn/basic_layers.py]: SyncBN does not have FP16 support for both training and inference. But for FP16 inference, SyncBN can be replaced with nn.BatchNorm as they have similar functionality.

**Pull request merged**